### DISCUSSION TITLE
A way to use SWR cache instead of Next one

### DISCUSSION THREAD

**Discussion by spacepolice10** [vercel/next.js#51679] [Help]
### Summary

In 12 you can return no data from getStaticProps and it will make app ignore SSR. So, if you have cached data in SWR you don't have to wait SSR, you might just use the data you already have. Is there a hack like that in 13 version, or should I go back with Pages Router?

**Comment by welschmoor** [comment]
I think in the old router you still get SSR even if you don't use.

Source: here in old router you can see, the data that you have (hard coded into html elements) is already inside the HTML that is being sent to the browser: https://www.youtube.com/watch?v=keP1PygtJ8c&list=PLC3y8-rFHvwgC9mj0qv972IO5DmD-H0ZH&index=16 

SWR is client side cache for things you fetch on the client. You can put data you fetch on the server inside SWR, I think it is called initialData or fallbackData. You can even disable client side fetching and only use this fallback data.

If you only get data from the server and make no fetching for client, you should avoid using SWR or ReactQuery

**Reply by spacepolice10** [reply]
Probably, I have stated what I wish to achieve incorrectly. In the old router, you can use getInitialProps which fires on the server only ONCE. Then you can check if window context exists and skip getInitialProps fire. So, you can spend no time on unnecessary fetch and use the data that you already have in SWR, redux or whatever the tool in use. App router does not let developers decide if they want to launch fetch or not. I wish I could make a decision and skip fetching data on route change when I already have all the necessary data in some form of client cache
